ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -570,4 +590,165 @@ private void reportUnrecognizedDirs() {
+      if (fileChannel.isOpen()) {
+        try {
+          fileChannel.close();
+        } catch (IOException e) {
+          //ensures that any error code won't be overwritten thus far
+          if (successfulThusFar) {
+            diskHealthStatus = diskHealthStatus.DELETE_IO_EXCEPTION;
+          }
+          logger.error("Exception occurred when deleting a file for disk healthcheck", e);
+          return false;
+        }
+      }
+      return true;
+    }
+
+    /**
+     * Will perform either a read or write operation on the disk while checking for exceptions and canceling the
+     * operation if there was an exception
+     * @param fileChannel used to perform the read/write operation
+     * @param buffer either used to write from or read to the buffer in the respective operation
+     * @param timeoutException indicate if the timeout is from reading or writing
+     * @param otherException indicate if the other exceptions are from reading or writing
+     * @param isWriting inform the function to either write or read with the buffer and filechannel
+     * @return True/False to indicate if the operation completed with no issues or not
+     */
+    private boolean performOperation(AsynchronousFileChannel fileChannel, ByteBuffer buffer,
+        DiskHealthStatus timeoutException, DiskHealthStatus otherException, boolean isWriting) {
+      Future<Integer> result = null;
+      try {
+        if (isWriting) {
+          result = fileChannel.write(buffer, 0);
+        } else {
+          result = fileChannel.read(buffer, 0);
+        }
+        result.get(operationTimeoutSeconds, TimeUnit.SECONDS);
+        return true;
+      } catch (TimeoutException e) {
+        diskHealthStatus = timeoutException;
+        logger.error("Timeout Exception occurred when operating on a file for disk healthcheck", e);
+      } catch (Exception e) {
+        diskHealthStatus = otherException;
+        logger.error("Exception occurred when operating on a file for disk healthcheck", e);
+      }
+      if (result != null) {
+        result.cancel(true);
+      }
+      return false;
+    }
+
+    /**
+     * creates the directory to store the file if it doesn't already exist and
+     * creates filechannel used to perform the disk healthcheck read/write operations
+     * @return filechannel if everything functioned without errors or null if there was an exception
+     */
+    private AsynchronousFileChannel createFileChannel() {
+      try {
+        File dir = new File(healthcheckDir);
+        if (!dir.exists() && !dir.mkdir()) {
+          throw new Error(String.format("%s directory wasn't created", healthcheckDir));
+        }
+        Path path = Paths.get(healthcheckDir, "temp");
+        AsynchronousFileChannel fileChannel =
+            AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.READ,
+                StandardOpenOption.DSYNC, StandardOpenOption.CREATE, StandardOpenOption.DELETE_ON_CLOSE);
```

Comment:
let's not use DSYNC for now, it might impact the disk performance.
At least, let's use a configuration here so we can easily turn this on and off.

Reply:
The amount of data written is really small. It won't affect disk perf. Either we buffer the data/metadata and flush it on close or do a direct_io. Without either of these, the test is incorrect.

Reply:
Included the SYNC flag now